packages/core/src/Overlay.vue

```diff
@@ -5,7 +5,7 @@ const importMetaUrl = isClient ? new URL(import.meta.url) : {}
 const protocol = inspectorOptions.serverOptions?.https ? 'https:' : importMetaUrl?.protocol
 const hostOpts = inspectorOptions.serverOptions?.host
 const host = hostOpts && hostOpts !== true ? hostOpts : importMetaUrl?.hostname
-const port = inspectorOptions.serverOptions?.port ?? importMetaUrl?.port
+const port = hostOpts && hostOpts !== true ? inspectorOptions.serverOptions?.port : importMetaUrl?.port
```

Comment:
When `hostOpts &&hostOpts! = = true `, `inspectorOptions serverOptions?.port` Is it possible is empty? If so, we may need to set a default value (importMetaUrl? .port) to make sure it works.